### DISCUSSION TITLE
I could use an insight with my codes.

### DISCUSSION THREAD

**Discussion by ckdlskgkdrhd** [naverz/zepeto-studio-global#37] [General]
heyall
I've been trying to create a false floor  and I'm facing a problem. 
could definitely use a help.

![dzdz](https://user-images.githubusercontent.com/110361648/184269595-6f62996f-1688-4fca-b995-aa29a559230b.PNG)

so basically, if a player steps on the trigger, the floor falls down from kinetic being turned off.
and it destroys by itself in the x time.
and then the instantiate runs TWICE despite having the line written on the triggerexit.
I don't get it. what am I doing wrong here.

**Comment by naverz-mkao** [comment]
Hi!

Your OnTriggerExit may be called more that once depending on if it is also touching some other collider object besides the character.

To fix that, you will want to do the same "if" check inside OnTriggerExit() that you did in OnTriggerEnter(). You will also probably want to disable the collider as well since there could be some cases where the platform could collide with the character more than once (depending on how you set up you level design).

You will then need to reenable the collider and set the rigidbody.isKinematic back to true before you call Instantiate so that the false platform will behave correctly if you jump on that false platform again.

Here's the code below with those changes. Hope this helps!

```import { Collider, Collision, CollisionDetectionMode, Debug, GameObject, Input, KeyCode, Rigidbody, Vector3, WaitForSeconds } from 'UnityEngine'
import { ZepetoPlayers } from 'ZEPETO.Character.Controller';
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class FalseFloor extends ZepetoScriptBehaviour {
    public floorPrefab:GameObject;
    public spawnPos:Vector3;
    public rb:Rigidbody;
//=========================ADD THIS===========================
    public collider:Collider;
//=========================ADD THIS===========================

    public LocalPP:GameObject;

    Start() {    
        this.spawnPos = this.transform.position;
    }

    OnTriggerEnter(col:Collision) {
        this.LocalPP = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.gameObject;
        if (col.gameObject != this.LocalPP) {
            return;
        }
        this.rb.isKinematic = false;
    }

    OnTriggerExit(col:Collision) {
//=========================ADD THIS===========================
        this.LocalPP = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.gameObject;
        if (col.gameObject != this.LocalPP) {
            return;
        }
        this.collider.enabled = false;
//=========================ADD THIS===========================

        this.StartCoroutine(this.TestRoutine());
        GameObject.Destroy(this.gameObject, 2.1);
        console.log('da thing deleted');
    }

    *TestRoutine() {
        yield new WaitForSeconds(2);

//=========================ADD THIS===========================
        this.rb.isKinematic = true;
        this.collider.enabled = true;
//=========================ADD THIS===========================
        var go:GameObject = GameObject.Instantiate(this.floorPrefab) as GameObject;

        console.log('da thing spawned');
        go.transform.position = this.spawnPos;
    }

}

**Reply by ckdlskgkdrhd** [reply]
thanks alot, I just had to restructure the script a bit after incorporating the checks.
